### PR TITLE
Move Cargo onto stable Rust

### PR DESCRIPTION
**Pull request by alexcrichton**
This commit moves Cargo onto 100% stable Rust now that the necessary fs features
have been stabilized. The tests were updated to understand when they're running
with a non-nightly compiler and disable all plugin-related tests. The only major
feature here is that the calls to `fs::set_file_times` were reimplemented
manually in the tests that Cargo has.

Cargo still requires a *nightly compiler* due to the stable features not having
made their way into the stable channel yet, but it will soon work on all of the
stable, beta, and nightly compilers once 1.1 is released!

### CODE REVIEW

File: tests/support/paths.rs [hunk 1]
```diff
@@ -68,37 +76,116 @@ impl CargoPathExt for Path {
...
+                    Ok(())
+                } else {
+                    Err(io::Error::last_os_error())
+                }
+            }
+        }
+
+        #[cfg(windows)]
+        fn set_file_times(p: &Path, atime: u64, mtime: u64) -> io::Result<()> {
+            extern crate kernel32;
```
Comment (retep998) [inline tests/support/paths.rs]: :smiling_imp:

File: tests/tests.rs [hunk 2]
```diff
@@ -58,3 +58,7 @@ mod test_shell;
 fn rustc_host() -> String {
     cargo::ops::rustc_version("rustc").unwrap().1
 }
+
+fn is_nightly() -> bool {
+    cargo::ops::rustc_version("rustc").unwrap().0.contains("-nightly")
```
Comment (chris-morgan) [inline tests/tests.rs]: Would not a development build (`-dev`) would satisfy the intent of the `is_nightly` check also (*viz.* using unstable features)?

**Comment by brson** [comment]
r=me with @chris-morgan's comment addressed.

**Pull request merged**